### PR TITLE
Add share button and modal for skill map activities

### PR DESCRIPTION
**Pull request by shakao**
skillmap-share

right now i have "share" visible on any activity that has a headerId (inprogress or completed). we don't actually load the tutorial state back in if you share a project that is mid-tutorial-step, probably because we've never had the ability to share projects mid-tutorial before... not sure if that's going to be a problem

### CODE REVIEW

File: skillmap/src/components/AppModal.tsx [hunk 1]
```diff
@@ -181,6 +200,59 @@ export class AppModalImpl extends React.Component<AppModalProps> {
...
         </Modal>
     }
+
+    protected handleShareInputClick = (evt: any) => { evt.target.select() }
+    protected handleShareCopyClick = () => {
+        const input = document.querySelector(".share-input input") as HTMLInputElement;
+        if (input) {
+            input.select();
+            document.execCommand("copy");
+            }
```
Comment (jwunderl) [inline skillmap/src/components/AppModal.tsx:210]:
```suggestion
        }
```

spacing nit

**Review by jwunderl — APPROVED** [review]

File: skillmap/src/components/AppModal.tsx [hunk 2]
```diff
@@ -181,6 +200,59 @@ export class AppModalImpl extends React.Component<AppModalProps> {
...
+        return <Modal title={resetModalTitle} actions={actions} onClose={this.handleOnClose}>
+            {data?.shortId ?
+                <div>{ lf("Your project is ready! Use the address below to share your projects.") }</div> :
+                <div className="share-disclaimer">
+                    { lf("You need to publish your project to share it or embed it in other web pages. You acknowledge having consent to publish this project.") }
+                </div>
+            }
+            {loading && <div className="share-loader">
+                <div className="ui active inline loader" />
+                <span>Loading...</span>
```
Comment (riknoll) [inline skillmap/src/components/AppModal.tsx:243]: lf?

**Review by riknoll — APPROVED** [review]

File: skillmap/src/components/AppModal.tsx (hunk 2 above)
Comment (livcheerful) [inline skillmap/src/components/AppModal.tsx:243]: Does this one need to be lf'ed?

**Review by livcheerful — APPROVED** [review]

File: skillmap/src/components/AppModal.tsx (hunk 2 above)
Comment (livcheerful) [reply skillmap/src/components/AppModal.tsx:243]: AH sorry i just saw richards comment >>

**Pull request merged**